### PR TITLE
feat: [PIE-9109]: Added support for UserMarkedFailure at stage level

### PR DESCRIPTION
**Pull request by Sagar-Rajput7**
### Summary

- Added Support for User Marked Failures (action) for Stage level
- Added User Marked Failure as Failure Strategy
- Added UserMarkedFailure settings support on the Settings page.

#### Screenshots

https://user-images.githubusercontent.com/98325173/230874279-47d0a855-2070-454e-8a6b-13ae24cfec9c.mov

Settings: 

Screenshot 2023-04-10 at 2 17 41 PM

Failure strategy:

Screenshot 2023-04-10 at 3 10 11 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by satyam-harness — APPROVED** [review]

File: src/modules/70-pipeline/strings/strings.en.yaml [hunk 1]
```diff
@@ -628,6 +630,8 @@ execution:
   dialogMessages:
     abortExecution: Are you sure you want to abort the execution?
     abortTitle: Abort Execution?
+    markAsFailedTitle: Mark Stage as Failed?
+    markAsFailedConfirmation: Are you sure you want to mark this Stage as failed? Doing so will mark this stage as failed and trigger the respective failure strategy.
```
Comment (PratyushGar) [inline src/modules/70-pipeline/strings/strings.en.yaml:634]:
```suggestion
    markAsFailedConfirmation: Are you sure you want to mark this stage as failed? Doing so will mark this stage as failed and trigger the respective failure strategy.
```

File: src/modules/70-pipeline/strings/strings.en.yaml [hunk 2]
```diff
@@ -628,6 +630,8 @@ execution:
   dialogMessages:
     abortExecution: Are you sure you want to abort the execution?
     abortTitle: Abort Execution?
+    markAsFailedTitle: Mark Stage as Failed?
```
Comment (PratyushGar) [inline src/modules/70-pipeline/strings/strings.en.yaml:633]:
```suggestion
    markAsFailedTitle: Mark stage as failed?
```

File: src/modules/23-default-settings/strings/strings.en.yaml [hunk 3]
```diff
@@ -12,3 +12,4 @@ pipelineTimeout: Pipeline Timeout
 stageTimeout: Stage Timeout
 stepTimeout: Step Timeout
 concurrentActivePipelineExecutions: Concurrent Active Pipeline Executions
+allowUserToMarkStepAsFailedExplicitly: Allow user to mark Step as failed explicitly
```
Comment (chetanvekariya-harness) [inline src/modules/23-default-settings/strings/strings.en.yaml:15]: Is explicitly needed at the end of sentence? If not required by PM then I think `Allow user to mark Step as failed explicitly` enough, I guess.

File: src/modules/70-pipeline/strings/strings.en.yaml [hunk 4]
```diff
@@ -508,6 +508,7 @@ failureStrategies:
     ProceedWithDefaultValues: Proceed with Default Values
     StepGroupRollback: Rollback Step Group
     MarkAsFailure: Mark As Failure
+    UserMarkedFailure: Mark Stage As Failed
```
Comment (Sagar-Rajput7) [inline src/modules/70-pipeline/strings/strings.en.yaml:511]: Title popover for the action button

File: src/modules/23-default-settings/strings/strings.en.yaml (hunk 3 above)
Comment (Sagar-Rajput7) [reply src/modules/23-default-settings/strings/strings.en.yaml]: I will follow up with the PM and BE on this (as this is in sync with the ID sent by BE - `allow_user_to_mark_step_as_failed_explicitly` )

File: src/modules/70-pipeline/strings/strings.en.yaml (hunk 4 above)
Comment (PratyushGar) [reply src/modules/70-pipeline/strings/strings.en.yaml:511]: For the title can you check with docs/PM, I still feel we should follow running case

File: src/modules/70-pipeline/strings/strings.en.yaml [hunk 5]
```diff
@@ -607,6 +608,7 @@ execution:
     abortedMessage: Stage "{{stageName}}" has been aborted successfully
     pausedMessage: Stage "{{stageName}}" execution will pause once the currently running steps have completed.
     resumedMessage: Stage  "{{stageName}}" has been resumed successfully
+    userMarkFailedMessage: Stage "{{stageName}}" has been marked failed.
```
Comment (PratyushGar) [inline src/modules/70-pipeline/strings/strings.en.yaml:611]: Nit:
```suggestion
    userMarkFailedMessage: Stage "{{stageName}}" has been marked failed
```
This is a phrase

**Review by PratyushGar — APPROVED** [review]

**Review by satyam-harness — APPROVED** [review]

**Pull request merged**